Speeds up world init. (93s -> 49s)

MrsTonedOne opened the pull request:
:cl:
tweak: world initialization is now faster.
/:cl:

* Armor is now new inited for obj and the first level of subpaths.
* Actions is now lazyinited and deleted with empty.
* Actiontypes is now only inited when actually used and deleted once it pre-fills actions with the action buttons.
* Pipes now prefill their node list(s) in new() using new /list/ (count) syntax to speed up the list initaliztions and remove the init proc.
* Pipes no longer store their item version, instead creating it on the fly when deconned
* Walls no longer store their metal stacks, instead creating it on the fly when deconned.
* obj, walls, floor, plating, item, machinery, structure, pipe, pipenet, atom, and movable no longer have an (init) proc. (along with a few other smaller examples)
* Atmos can pass checking is now a var with the ability to have a proc be called in advance cases.
  * (as a side effect, I had to fix a few things that were calling atmosCanPass rather then using the pre-calculated list, this should speed up chemfoam and flame effects greatly)
* Fixed turfs smoothing twice at world init.  (please smack down anybody using `smooth_icon` and not `queue_smooth`) (this alone shaved off 8 seconds)
* Made smoothing batch overlay updates to lower on appearance churn (this alone shaved off 5 seconds)

Top is old, bottom is new (ruin generation disabled to avoid noise)
![image](https://cloud.githubusercontent.com/assets/7069733/20990712/19f789cc-bc8f-11e6-8ff7-54d54ac4c0d4.png)

(93 seconds -> 49 seconds)

We are hitting the point of diminishing returns. there are now less big ticket things to fix, and instead a whole lot of small ticket things to fix.

MrsTonedOne committed:
Speeds up world init.

* Armor is now new inited for obj and the first level of subpaths.
* Actions is now lazyinited and deleted with empty.
* Actiontypes is now only inited when actually used and deleted once it pre-fills actions with the action buttons.
* Pipes now prefill their node list(s) in new() using new /list/ (count) syntax to speed up the list initaliztions and remove the init proc.
* Pipes no longer store their item version, instead creating it on the fly when deconned
* Walls no longer store their metal stacks, instead creating it on the fly when deconned.
* obj, walls, floor, plating, item, machinery, structure, pipe, pipenet, atom, and movable no longer have an (init) proc. (along with a few other smaller examples)
* Atmos can pass checking is now a var with the ability to have a proc be call in advance cases.
  * (as a side effect, I had to fix a few things that were calling atmosCanPass rather then using the pre-calculated list, this should speed up chemfoam and flame effects greatly)

PKPenguin321 commented:
>Pipes no longer store their item version, instead creating it on the fly when deconned
Walls no longer store their metal stacks, instead creating it on the fly when deconned.

what the heck? im surprised this never caused issues

MrStonedOne commented:
it was added when HG buffed the sing to make the sing faster, but I decided it was lame and removed it.

Uses up extra memory for no reason, adds initializing these to world init, etc.

Mervill commented:
And you still managed to remove more lines then you added. Good show!

MimicFaux commented:
That's the sign of clean(er) code.

TechnoAlchemisto commented:
I'd argue not to sacrifice singularity smoothness but the song is such a rare sight as it is it isn't really a problem 👍🏻

RemieRichards commented on code/modules/client/client_procs.dm:
```diff
@@ -1,7 +1,7 @@
 	//SECURITY//
 	////////////
-#define UPLOAD_LIMIT		1048576	//Restricts client uploads to the server to 1MB //Could probably do with being lower.
+#define UPLOAD_LIMIT		100048576	//Restricts client uploads to the server to 1MB //Could probably do with being lower.
```
hmm?

MrStonedOne replied:
oh rip, good catch.. I was proc calling load_map with the station map files to profile world init. they are bigger than 1mb for uploading a file for the arg for advance proc call

MrsTonedOne committed:
Reverts upload limit
(remind me one day to defuck this, it could easily be a config thats not editable by vv to make changes easier)

MrsTonedOne committed:
Makes apc update icon a bit faster.
APC new is some what high on the profile of world init, still not sure why, but this stood out as a waste of cpu so i fixed it.

MrsTonedOne committed:
Fixes runtime with atmos backpack water tanks.

MrStonedOne commented:
Got more speedups coming. I figured out more tricks to profile world init accurately.

MrsTonedOne committed:
Makes smoothing faster (and fixes turfs smoothing twice at init)

MrsTonedOne committed:
Makes apcs init faster by replacing some spawns with addtimer

MrStonedOne commented:
93 seconds to 49 seconds now.

MrStonedOne commented:
do not merge, issue found.

MrsTonedOne committed:
fix transit turfs.
